### DISCUSSION TITLE
return vectors field error

### DISCUSSION THREAD

**Discussion by arvissugrrr** [milvus-io/milvus#28009] [Q&A and General discussion]
Hi, 

I created a db as belows

uid, int64
context, VARCHAR
vecs, FLOAT_VECTOR

I use search with IVF_FLAT or IVF_PQ or others when set up output_fileds = ['vecs'] will get error
like this 
"query failed: vector output fields for IVF_FLAT index with COSINE metric is not allowed:"

![image](https://github.com/milvus-io/milvus/assets/114966498/db6e2c80-14e6-459b-8a07-6c01794a2f26)

and use FLAT is ok. 

![image](https://github.com/milvus-io/milvus/assets/114966498/591bff04-f524-45c2-b27c-88624d71472b)

so just FLAT support return vecs field?

**Comment by xiaofan-luan** [comment]
IVF_FLAT should work, HNSW should work too.

For 2.3.2 all index types should be supported

**Comment by arvissugrrr** [comment]
my milvus version is 2.3.1, 
can work : FLAT, HNSW, SCANN 
cant : IVF_FLAT, IVF_SQ8, IVF_PQ 

![image](https://github.com/milvus-io/milvus/assets/114966498/160e2053-8ed4-4ed6-8cfd-d9b5cf602e20)

**Comment by xiaofan-luan** [comment]
Milvus 2.3.2 should be able to solve the problem?

But you need to setup a localdisk to hold the data
